src/ngx_http_lua_connection.c

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (C) 2014-2015 Daurnimator
+ */
+
+/* need to include DDEBUG so that ngx_http_lua_util.h works */
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+#include "ngx_http_lua_connection.h"
+
+#include <math.h> /* HUGE_VAL */
+#include <poll.h> /* POLLIN, POLLOUT */
```

Comment:
FWIW, I just looked it up, and NGX_READ_EVENT is a #define to POLLIN on linux+osx anyway.

The problem is that NGX_READ_EVENT is not safe to be used as a bitmask.
e.g. see the `select` `#ifdef`:
``` C
#define NGX_READ_EVENT     0
#define NGX_WRITE_EVENT    1
```